atom/browser/lib/rpc-server.coffee

```diff
@@ -10,6 +10,7 @@ valueToMeta = (sender, value) ->
   meta.type = 'buffer' if Buffer.isBuffer value
   meta.type = 'value' if value is null
   meta.type = 'array' if Array.isArray value
+  meta.type = 'promise' if Promise.resolve(value) == value
```

Comment:
You are actually creating a new promise for that even if it's not a promise, It would be ideal to do something like
```
value && value.constructor.name === 'Promise'
// or maybe
value instanceof Promise
```